python/ray/dataframe/test/test_dataframe.py

```diff
@@ -2205,18 +2209,14 @@ def test_pow():
     test_inter_df_math("pow", simple=False)
 
 
-def test_prod():
-    ray_df = create_test_dataframe()
-
-    with pytest.raises(NotImplementedError):
-        ray_df.prod(None)
-
+@pytest.fixture
+def test_prod(ray_df, pandas_df):
+    assert(ray_df_equals_pandas(ray_df.prod().sort_index(), pandas_df.prod().sort_index()))
```

Comment:
Please remove `sort_index()` from each side of this. The index should come out the same.